### PR TITLE
Add sync-interval flag

### LINKED ISSUE #18: Sync service ticker is hardcoded
The ticker that refresh available new version is now currently hardcoded, but this should be at least bumped to a higher value, and offer a way to optionally override this.

Ideally, we would also like to trigger the service as soon as a `ManagedOSVersionChannel`  has been updated.

Follow up #14

### PR DESCRIPTION
**Pull request by Itxaka**
Fixes #18 

Signed-off-by: Itxaka <igarcia@suse.com>

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -31,6 +31,7 @@ import (
 var (
 	namespace  = flag.String("namespace", "cattle-rancheros-operator-system", "Namespace of the pod")
 	namespaces = flag.String("namespaces", "", "A comma separated list of namespaces to watch")
+	interval   = flag.Int("sync-interval", 10, "Interval (in seconds) for the upgrade channel ticker")
```
Comment (mudler) [inline main.go]: I'd go here for a string and use `time.ParseDuration`, which directly returns a duration. In this way the user can supply things like "120m", "10s", etc.

Comment (kkaempf) [reply main.go]: Not sure how this matches with customer's "maintenance windows", which might happen once a week / month ?!
Do we have documentation how this fits into the larger "image update" workflow ?

Comment (mudler) [reply main.go]: my bad for having used _user_ there. We are going to have a proper default and this shouldn't really be something customizable. We have to expose it to the chart first

Comment (davidcassany) [reply main.go]: In addition I would use a default that is _suitable_ in a real environment (1 or 2h maybe?) and shorten it for our tests.

Comment (Itxaka) [reply main.go]: That parseduration sounds pretty good, I didn't knew that.

Comment (Itxaka) [reply main.go]: wait, so we want this to be exposed or not?? Because if its in the chart values, then its customizable???

Comment (Itxaka) [reply main.go]: I updated to expose it in the chart, not sure what the confusion is here to be honest

**Review by davidcassany — APPROVED** [review]
Looks good

**Comment by Itxaka** [comment]
@mudler @kkaempf any concerns with this? Maybe its missing getting the value from the ENV as I think that is usually one of the principles of pods, getting stuff from env vars in the pod deployment?

**Comment by mudler** [comment]
> @mudler @kkaempf any concerns with this? Maybe its missing getting the value from the ENV as I think that is usually one of the principles of pods, getting stuff from env vars in the pod deployment?

that's fine, I've other changes on #22 that are introducing a basic skeleton for the CLI. this looks good for the moment

**Review by mudler — APPROVED** [review]

**Pull request merged**